### PR TITLE
Integration Tests for Node Topologies

### PR DESCRIPTION
**Pull request by harshitap26**
# Description

- PR to add Integration tests for adding and passing zone labels through the topologies.
- Made changes to the function arguments for reading array config since we are using different config for the zoning tests.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1612 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Integration tests successfully running for the added scenarios.
```
  Scenario: call NodeGetInfo                           # features/integration.feature:1009
    Given a VxFlexOS service                           # <autogenerated>:1 -> *feature
[iCallNodeGetInfo] Calling NodeGetInfo...
[nodeGetInfo] Calling NodeGetInfo...
INFO[0005] failed to probe node: rpc error: code = FailedPrecondition desc = DrvCfgQuerySystems: Request to query MDM failed : exit status 125 
DEBU[0005] Node labels: map[kubernetes.io/hostname:node1 zone.csi-vxflexos.dellemc.com:zoneA] 
DEBU[0005] MaxVolumesPerNode: 0                         
INFO[0005] Arrays: map[205f819048a7000f:0xc0003205b0]   
INFO[0005] Probing all arrays. Number of arrays: 1      
INFO[0005] default array is set to array ID: 205f819048a7000f 
INFO[0005] 205f819048a7000f is the default array, skipping VolumePrefixToSystems map update.  
INFO[0005] array 205f819048a7000f probed successfully   
INFO[0005] Zone found for node ID: , adding system ID: 205f819048a7000f to node topology 
DEBU[0005] NodeId: D2802C64-F3A7-4B5A-9035-6FEC669A2FB0 
    And I call NodeGetInfo with ""                     # <autogenerated>:1 -> *feature
2024/12/05 12:10:08 Node Accessibility segments:<key:"csi-vxflexos.dellemc.com/205f819048a7000f" value:"csi-vxflexos.dellemc.com" > segments:<key:"zone.csi-vxflexos.dellemc.com" value:"zoneA" > 
    Then a NodeGetInfo is returned with zone topology  # <autogenerated>:1 -> *feature
2024/12/05 12:10:08 Node Accessibility segments:<key:"csi-vxflexos.dellemc.com/205f819048a7000f" value:"csi-vxflexos.dellemc.com" > segments:<key:"zone.csi-vxflexos.dellemc.com" value:"zoneA" > 
array found  205f819048a7000f map[string]interface {}{"allSystemNames":"", "endpoint":"https://10.247.102.220", "insecure":true, "isDefault":true, "nasName":"", "password":"********", "systemID":"205f819048a7000f", "user":"admin"}
2024/12/05 12:10:08 array systemID 205f819048a7000f
    And a NodeGetInfo is returned with system topology # <autogenerated>:1 -> *feature
=== RUN   TestZoneIntegration/call_NodeGetInfo#01

  Scenario: call NodeGetInfo                                                # features/integration.feature:1016
    Given a VxFlexOS service                                                # <autogenerated>:1 -> *feature
[iCallNodeGetInfo] Calling NodeGetInfo...
[nodeGetInfo] Calling NodeGetInfo...
INFO[0005] failed to probe node: rpc error: code = FailedPrecondition desc = DrvCfgQuerySystems: Request to query MDM failed : exit status 125 
DEBU[0005] Node labels: map[invalid_zone_key:zoneA kubernetes.io/hostname:node1] 
DEBU[0005] MaxVolumesPerNode: 0                         
INFO[0005] Arrays: map[205f819048a7000f:0xc0003205b0]   
INFO[0005] Probing all arrays. Number of arrays: 1      
INFO[0005] default array is set to array ID: 205f819048a7000f 
INFO[0005] 205f819048a7000f is the default array, skipping VolumePrefixToSystems map update.  
INFO[0005] array 205f819048a7000f probed successfully   
INFO[0005] No zoning found for node ID: , adding system ID: 205f819048a7000f 
DEBU[0005] NodeId: D2802C64-F3A7-4B5A-9035-6FEC669A2FB0 
    And I call NodeGetInfo with "invalid_zone_key"                          # <autogenerated>:1 -> *feature
2024/12/05 12:10:08 Node Accessibility segments:<key:"csi-vxflexos.dellemc.com/205f819048a7000f" value:"csi-vxflexos.dellemc.com" > 
    Then a NodeGetInfo is returned without zone topology "invalid_zone_key" # <autogenerated>:1 -> *feature

4 scenarios (4 passed)
21 steps (21 passed)
854.519865ms
INFO[0005] gracefully stopped                           
--- PASS: TestZoneIntegration (5.86s)
    --- PASS: TestZoneIntegration/Create_publish,_unpublish,_and_delete_zone_volume (0.79s)
    --- PASS: TestZoneIntegration/Create_zone_volume_with_invalid_zone_information (0.00s)
    --- PASS: TestZoneIntegration/call_NodeGetInfo (0.04s)
    --- PASS: TestZoneIntegration/call_NodeGetInfo#01 (0.01s)
PASS
coverage: [no statements]
ok      github.com/dell/csi-vxflexos/v2/test/integration        5.900s
```
- [x] Tested the existing scenarios for NFS provisioning to verify the changes made to the function arguments of reading the config file 
[integration-test.log](https://github.com/user-attachments/files/18035962/integration-test.log)

### CODE REVIEW

File: test/integration/integration_test.go [hunk 1]
```diff
@@ -31,21 +31,22 @@ import (
...
 
 var grpcClient *grpc.ClientConn
 
-func init() {
+func readConfigFile(configurationFile string) {
 	/* load array config and give proper errors if not ok*/
-	if _, err := os.Stat(configFile); err == nil {
-		if _, err := os.ReadFile(configFile); err != nil {
-			err = fmt.Errorf("DEBUG integration pre requisites missing %s with multi array configuration file ", configFile)
+	if _, err := os.Stat(configurationFile); err == nil {
```
Comment (donatwork) [inline test/integration/integration_test.go:44]: This seems like an unnecessary change. It's a common idiom to use config to mean configuration.

File: test/integration/integration_test.go [hunk 2]
```diff
@@ -118,10 +120,11 @@ func TestIntegration(t *testing.T) {
 }
 
 func TestZoneIntegration(t *testing.T) {
+	readConfigFile(zoneConfigFile)
 	var stop func()
 	ctx := context.Background()
 	fmt.Printf("calling startServer")
-	grpcClient, stop = startServer(ctx, "features/array-config/multi-az")
+	grpcClient, stop = startServer(ctx, zoneConfigFile)
```
Comment (donatwork) [inline test/integration/integration_test.go:127]: And here you used zoneConfigFile instead of zoneConfigurationFile which is now inconsistent with your other change.

File: test/integration/step_defs_test.go [hunk 3]
```diff
@@ -103,7 +105,7 @@ func (f *feature) getGoscaleioClient() (client *goscaleio.Client, err error) {
 	if f.arrays == nil {
 		fmt.Printf("Initialize ArrayConfig from %s:\n", configFile)
 		var err error
-		f.arrays, err = f.getArrayConfig()
+		f.arrays, err = f.getArrayConfig(configFile)
```
Comment (donatwork) [inline test/integration/step_defs_test.go:108]: In this file you now mix configFile and configurationFile.

**Review by donatwork — APPROVED** [review]

File: test/integration/integration_test.go (hunk 1 above)
Comment (harshitap26) [reply test/integration/integration_test.go:44]: @donatwork Since we have introduced a new config file for the multi-az Integration tests and we want to reuse the same function (readConfigFile) for both integration tests, so we are now passing the config file as the argument, so I have used the argument name as configurationFile. 

configFile is constant that still points to the path of the config file.

File: test/integration/step_defs_test.go (hunk 3 above)
Comment (harshitap26) [reply test/integration/step_defs_test.go:108]: Here we are calling the function, hence passing the path of the config file.

Comment (harshitap26) [reply test/integration/step_defs_test.go:108]: We are consistent with passing configFile and zoneConfigFile paths as arguments when calling the getArrayConfig or readConfigFile functions.

**Review by falfaroc — APPROVED** [review]
